tests/CMakeLists.txt

```diff
@@ -383,8 +383,8 @@ add_test(NAME validate-secondary-interface
     --firmware ${PROJECT_SOURCE_DIR}/tests/test_data/firmware.txt)
 
 
-add_subdirectory(sota_tools)
 if(BUILD_SOTA_TOOLS)
+    add_subdirectory(sota_tools)
```

Comment:
Don't you still want to add `sota_tools` even if not building them so that the source files get checked? Or do we not care about that for the tests?

Reply:
Yes, that's a fair point. We don't enforce source checks in `tests` right now but I'll move the condition inside the directory for consistency.